syntax = "proto2";

import "google/protobuf/descriptor.proto";

enum EINT
{

    INT8  = 0;
    INT16  = 1;
}
enum EUINT
{
    UINT8 = 0;
    UINT16 = 1;
}
extend google.protobuf.FieldOptions
{
    // max count for repeated
    optional uint32 max_count = 30000;
    // max count for repeated fixed
    optional uint32 fixed_max_count = 30001;
    // real int type
    optional EINT int_type = 30002;
    optional EUINT uint_type = 30003;
    // max len of string
    optional uint32 size = 30004;
    // gen pod ignore this field
    optional bool ignore = 30005 [default = false];
}

extend google.protobuf.MessageOptions
{
    // 1: need to generate pod data code
    // 2: need to generate pod data code without constructor
    optional bool gen_pod = 30000;
}

